gr-network/lib/tcp_sink_impl.cc

```diff
@@ -228,8 +228,6 @@ int tcp_sink_impl::work(int noutput_items,
                         gr_vector_const_void_star& input_items,
                         gr_vector_void_star& output_items)
 {
-    gr::thread::scoped_lock guard(d_setlock);
```

Comment:
It looks like this does protect some attributes. If I'm reading this Boostery correctly, `accept_handler()` runs async, and `run_listener()` runs in its own thread. The former modifies `d_connected` (could be made `std::atomic_bool`, or maybe even `std::atomic_flag`), `d_tcpsocket` (maybe we can use `d_setlock` just for this). The latter modifies `d_start_new_listener` (could also be properly using atomic types instead of requiring a setlock).

Reply:
`d_setlock` simply is not touched anywhere else but `stop`, so there's simply nothing protected here. gnuradio-runtime itself doesn't use it.

Reply:
Yeah you're right, the single lock is nonsense. But unlike the UDP blocks, I think we should be doing some safe concurrent accesses here.

If you remove the lock, then the code is just as good/bad as it is before. So sure, this change won't do any harm. But now that you've opened that can of worms, what about the missing locks in the TCP sink?

I guess you could say "it's unsafe already, and at least it'll be faster". Why not! Or, you could put locks in run_listener() and accept_handler(), too. ("it's safe now, albeit a bit slower"). Are you even supposed to put locks in an accept_handler function? Don't know! Brain still in holidays-mode.

Reply:
So, I opened that can of worms, and I stared into it. And the worms stared back: you're right, we *must* protect things from concurrent access from accept_handler / run_listerner. That's an oversight on my end.